src/distributed_join.cuh

```diff
@@ -460,12 +364,45 @@ distributed_inner_join(
         flag = false;
     }
 
+    /* Copy from hashed table to communicated table for the current rank */
```

Comment:
Can you add more details why we need - you explained on slack, but would be good to add motivation for moving the copies in here in the comments as well.

Reply:
Added in 4f7f099.